beaker-vue/src/components/cell/NextBeakerQueryCell.vue

```diff
@@ -173,66 +173,55 @@ watchEffect(() => {
     }
 });
 
-const badgeSeverity = computed(() => {
-    if (mockStickyForce.value) return 'info';
-
-    const queryStatus = cell.value.metadata?.query_status;
-
-    switch (queryStatus) {
-        case 'success':
-            return 'success';
-        case 'failed':
-            return 'danger';
-        case 'aborted':
-            return 'warn';
-        case 'in-progress':
-            return 'info';
-        case 'pending':
-        default:
-            return 'secondary';
-    }
-});
-
-const badgeIcon = computed(() => {
-    if (mockStickyForce.value) return 'pi pi-spinner pi-spin busy-icon';
-
-    const queryStatus = cell.value.metadata?.query_status;
-
-    switch (queryStatus) {
-        case 'success':
-            return 'pi pi-check bolded';
-        case 'failed':
-            return 'pi pi-times';
-        case 'aborted':
-            return 'pi pi-minus';
-        case 'in-progress':
-            return 'pi pi-spin pi-spinner busy-icon';
-        case 'pending':
-            return 'pi pi-clock';
-        default:
-            return null;
+const badgeData = computed(() => {
+    // Handle mock sticky mode first
+    if (mockStickyForce.value) {
+        return {
+            severity: 'info',
+            icon: 'pi pi-spinner pi-spin busy-icon',
+            tooltip: 'Mock sticky mode active'
+        };
     }
-});
-
-const badgeTooltip = computed(() => {
-    if (mockStickyForce.value) return 'Mock sticky mode active';
 
     const queryStatus = cell.value.metadata?.query_status;
+    const cellStatus = cell.value.status;
+    const isActive = ['busy', 'awaiting_input'].includes(cellStatus);
```

Comment:
Is there an edge case to this fix I missed?

I tested:

- Agent asking a question
- Agent working continuously/busy
- Agent manually marking as success
- User clicking abort and aborting query cell.